### ISSUE TITLE
LinearRegression should be a shorcut to polynomial_regression(degree = 1)

### ISSUE THREAD

**Issue by advy99** [advy99/cpp_ml#8]
With #7 there is no sense of mantaining both linear_regression and polynomial_regression, as linear_regression is a particular case of polynomial_regression with degree = 1

All the code of linear_regression should be deleted, and because Linear Regression is widely used, create a shorcut to polynomial_regression(degree = 1) with the name of linear_regression (maybe a typedef is enough??)

**Comment by frey0-0** [comment]
@advy99 I would like to do this issue. Basically I have to remove the linear regression code and replace linear regression call with polynomial regression call with argumanent 1 in the test.cpp right?

**Comment by advy99** [comment]
> @advy99 I would like to do this issue. Basically I have to remove the linear regression code and replace linear regression call with polynomial regression call with argumanent 1 in the test.cpp right?

Not exactly. linear_regression is a specific case of polynomial_regression (is just polynomial_regression with degree = 1), so i want to mantain the type `linear_regression` but this class should be just a wrapper to `polynomial_regression` methods, and not do the same math as the `polynomial_regression` class (currently is mostly duplicated code). 

I thing a good solution would be to rewrite `linear_regression` as a child class of `polynomial_regression` where degree is always one (this child class only need a default constructor to set the degree to one, the predict and fit methods are the polynomial_regression methods)

The test.cpp does not contain anything of linear/polynomial regression, so there is no need to touch it.

**Comment by frey0-0** [comment]
Oh nice, I haven't done anything in OOPs I would like to take this issue @advy99 :) then child class will be present in the polynomial_regression file only right?

**Comment by advy99** [comment]
> Oh nice, I haven't done anything in OOPs I would like to take this issue @advy99 :) 

Perfect! I will assign this issue to you.

> then child class will be present in the polynomial_regression file only right?

Not exactly. The file `include/regressors/linear_regression.hpp` should contain the current class, but must be a child class from `polynomial_regression` instead of `regressor`, and only have a public constructor (only this constructor, no other data members or methods) that set the degree to 1 (implemented in `src/regressors/linear_regression.cpp`). You also need to modify `include/regressors/polynomial_regression.hpp` and change the private section to protected, in order to have access to these members in the child class. Hope that helps!
